green/test/test_loader.py

```diff
@@ -264,6 +264,16 @@ def test_method2(self):
                          set(['test_method1', 'test_method2']))
 
 
+    def test_nose_disabled_attribute(self):
+        "Tests disabled by nose generators dont get loaded"
+        class HasDisabled(unittest.TestCase):
+            def test_method(self):
+                pass
+
+        HasDisabled.test_method.__test__ = False
```

Comment:
Okay, finally resolved this. Apparently unittest binds methods to the test class which means that [you can't access them like function objects outside of the test class](http://stackoverflow.com/questions/7891277/why-does-setattr-fail-on-a-bound-method).

The solution was just to add the `__test__` attribute within the class body before unittest has done any work.